spec/app/ics-020-fungible-token-transfer/README.md

```diff
@@ -311,26 +321,36 @@ function onRecvPacket(packet: Packet) {
        amount: packet.amount
      }
      tokens = []Token{token}
+     receiver = packet.receiver
   } else if transferVersion == "ics20-2" {
     FungibleTokenPacketDataV2 data = UnmarshalJSON(packet.data)
     tokens = data.tokens
+    // if we need to forward the tokens onward
+    // temporarily send to the channel escrow address of the intended receiver
+    if len(forwardingPath) > 0 {
+      receiver = channelForwardingAddresses[packet.destinationChannel]
```

Comment:
Could be ok. If we remove the `channelForwardingAddresses` and use `channelEscrowAddresses` we could do directly: 

`receiver = channelEscrowAddresses[packet.destinationChannel]`

Reply:
I still think that this should be:

```
nextPort, nextChannel = split(forwardingPath[0], "/")
channelEscrowAddresses[nextChannel]
```

because `packet.destinationChannel` is the same as the source channel where tokens where sent from, and therefore `channelEscrowAddresses[packet.destinationChannel]` would be equal to the escrow address where the tokens were locked when they left the chain. 

I hope this makes sense!

Reply:
Since we are using different addresses from the escrow this should be fine so long as we stay consistent. Do you see a bug in the logic, or just a preference?